### ISSUE TITLE
`calcDist` uses `getTransportCost` but really needs distance

### ISSUE THREAD

**Issue by pglotov** [graphhopper/jsprit#234]
In /jsprit/core/algorithm/ruin/distance/AvgServiceAndShipmentDistance.java:84 `private double calcDist(Location location_i, Location location_j)` calls `costs.getTransportCosts()`. Looks like what it wants is distance but calls for cost function with `vehicle = null` which feels inconsistent since cost generally depends on vehicle. Shouldn't it actually be using a distance calculator? Most cost calculators i've seen also have distance calculators in them. Seems like cost calculators should also implement a distance function `getDistance()` and `calcDistance` should be calling it. Does it sound reasonable?

**Comment by PGWelch** [comment]
Having logic in there which depends on a try-catch is a bit dodgy, but if the distance is being used as a metric to group 'close' jobs into clusters for a ruin, then in my opinion the 'distance' really should be transport cost, not actual distance.

**Comment by pglotov** [comment]
The thing is cost depends on vehicle. Which one should be used to compute "cost radius"? Also, in the [paper](http://www.staff.uni-mainz.de/schneidj/papers/ruinandrecreate.pdf) they seem to use distance rather than cost for radial ruin.

**Comment by PGWelch** [comment]
Hmm. I see your point about cost being dependent on vehicle. It may be more appropriate to calculate a mean vehicle cost per unit distance and cost per unit time across all vehicles and use it to generate the cost here? That way at least we're creating a cost averaged over all vehicles? 

Various academic papers optimise for minimising distance but in the real world its usually minimising travel time that is much more appropriate. Therefore just assuming an arbitrary cost - which can be a user-defined weighting of both distance and time, is safer in my opinion.

**Comment by pglotov** [comment]
Average cost would make sense. However, in [CrowFlyCost](https://github.com/graphhopper/jsprit/blob/1f4974c2106ffad9f6688a3b077e16d076c2a29f/jsprit-core/src/main/java/com/graphhopper/jsprit/core/util/CrowFlyCosts.java) the cost function actually returns distance if no vehicle is provided. May be we just need to call the distance function explicitly when doing ruin.

**Comment by oblonski** [comment]
Thanks for your thoughts. Average transport cost sounds reasonable. However, I think we do not need to be too sophisticated here. This is just a measure to define a neighborhood for radial ruin. The term `distance` here does not necessarily mean that we need to use 'm' or 'km'. It is just a general measure to define the neighborhood. It can also be 'time' or smth depending on the jobs' time windows etc..

**Comment by pglotov** [comment]
I agree "pure distance" is a good measure. It's just that it is obtained after a call to `getTransportCost` with `vehicle = null`. I think it would be better to call `getDistance()` directly, without using the obscure convention that when `vehicle` is `null` cost function should return distance.
